mongo/readconcern/readconcern.go

```diff
@@ -11,10 +11,17 @@
 package readconcern // import "go.mongodb.org/mongo-driver/mongo/readconcern"
 
 import (
+	"errors"
+
 	"go.mongodb.org/mongo-driver/bson/bsontype"
 	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
+// ErrEmptyReadConcern indicates that a read concern has no fields set.
+//
+// Deprecated: ErrEmptyReadConcern will be removed in Go Driver 2.0.
+var ErrEmptyReadConcern = errors.New("a read concern must have the level field set")
```

Comment:
In the proposed solution to throw this error on "rc == nil", a user could define an empty read concern (no level) and _not_ get this error. Perhaps it should say something like "cannot marshal nil value":

```suggestion
var ErrEmptyReadConcern = errors.New("cannot marshal nil value")
```

Reply:
We should not export `ErrEmptyReadConcern` because it does not support a known use case. See my other comment for details.